### PR TITLE
Replace cpprestsdk usage with libcurl

### PR DESCRIPTION
**Pull request by shishirb-MSFT**

### CODE REVIEW

File: client-lite/CMakeLists.txt [hunk 1]
```diff
@@ -33,16 +33,8 @@ endfunction ()
...
-# 1. v2.10.10 min version (see required PR link below). cpprestsdk does not seem to support specifying this through cmake.
-# https://github.com/microsoft/cpprestsdk/pull/1019/files
-#
-# 2. Installing libcpprest-dev via apt installs cpprest's cmake config files to a non-default cmake search path before v2.10.9
-# See: https://github.com/microsoft/cpprestsdk/issues/686
-# This issue has been patched but has not made its way to the stable branch for Ubuntu
-# Since we are statically linking to v2.10.16 we no longer need to worry about the above as cpprest is patched to provide the proper package configuration metadata
-find_package(cpprestsdk CONFIG REQUIRED)
+find_package(CURL REQUIRED)
+find_package(Threads REQUIRED)
```
Comment (jimson-msft) [inline client-lite/CMakeLists.txt:37]: nit: consider adding a comment here for which header this is required for

File: licenses/cgmanifest.json [hunk 2]
```diff
@@ -66,6 +56,42 @@
...
+          "Name": "libcurl4",
+          "Version": "7.64.0-4+deb10u2",
+          "Distribution": "Debian",
+          "Release": "10.10"
+        }
+      },
+      "DevelopmentDependency": false
+    },
     {
       "Component": {
```
Comment (jimson-msft) [inline licenses/cgmanifest.json:96]: just wondering - do we need to also include the threads library here?

**Review by jimson-msft — APPROVED** [review]
:shipit:

File: client-lite/CMakeLists.txt (hunk 1 above)
Comment (shishirb-MSFT) [reply client-lite/CMakeLists.txt:38]: Not any particular header but a requirement from g++ when we use any threads in the project.

File: licenses/cgmanifest.json (hunk 2 above)
Comment (shishirb-MSFT) [reply licenses/cgmanifest.json:96]: No idea. Is it considered a 3rd party lib? We don't list c/c++ libs here either.

**Pull request merged**